pkg/cmd/admin/policy/modify_roles.go

```diff
@@ -303,8 +305,21 @@ func (o *RoleModificationOptions) AddRole() error {
 	roleBinding.RoleRef.Namespace = o.RoleNamespace
 	roleBinding.RoleRef.Name = o.RoleName
 
-	roleBinding.Users.Insert(o.Users...)
-	roleBinding.Groups.Insert(o.Groups...)
+	newSubjects := authorizationapi.BuildSubjects(o.Users, o.Groups, uservalidation.ValidateUserName, uservalidation.ValidateGroupName)
+	newSubjects = append(newSubjects, o.Subjects...)
+
+subjectCheck:
```

Comment:
Style comment - never use goto continue labels.  Always turn it into a nested function.  Don't have to do it here, but it's basically just an indication you wanted to return early from a function.